modules/mytargetBidAdapter.js

```diff
@@ -0,0 +1,108 @@
+import * as utils from '../src/utils';
+import * as url from '../src/url';
+import { config } from '../src/config';
+import { registerBidder } from '../src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'mytarget';
+const BIDDER_URL = '//ad.mail.ru/hbid_prebid/';
+const DEFAULT_CURRENCY = 'RUB';
+const DEFAULT_TTL = 180;
+
+function buildPlacement(bidRequest) {
+  let { bidId, params } = bidRequest;
+  let { placementId, position, response, bidfloor } = params;
+  let placement = {
+    placementId,
+    id: bidId,
+    position: position || 0,
+    response: response || 0
+  };
+
+  if (typeof bidfloor !== 'undefined') {
+    placement.bidfloor = bidfloor;
+  }
+
+  return placement;
+}
+
+function getSiteName(referrer) {
+  let sitename = config.getConfig('mytarget.sitename');
+
+  if (!sitename) {
+    sitename = url.parse(referrer).hostname;
+  }
+
+  return sitename;
+}
+
+function generateRandomId() {
+  return Math.random().toString(16).substr(2);
```

Comment:
`substr` should be replaced with `substring`, you can view reason here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

Reply:
@vfedoseev, thank you!
I've committed the fixed version, but the tests on CircleCI failed.
https://circleci.com/gh/prebid/Prebid.js/1895

By the way, should the same replaement be applied to the core?
https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L87